File: js/task1/S1_VAR_06.js
function classifyNumberLike(x) {
  if (Number.isNaN(x)) {
      return "nan";
  }

  if (typeof x === 'number') {
      return "number";
  }

  return "not-a-number";
}

console.log(classifyNumberLike(NaN));         
console.log(classifyNumberLike(0));           
console.log(classifyNumberLike("0"));         
console.log(classifyNumberLike("abc"));      
console.log(classifyNumberLike(undefined));   
console.log(classifyNumberLike(null));

File: python/task1/S1_VAR_06.py
def classifyNumberLike(x):
  if x != x:
      return "nan"
  if isinstance(x, (int, float)):
      return "number"
  return "not-a-number"

print(classifyNumberLike(float('nan')))
print(classifyNumberLike(0))
print(classifyNumberLike("0"))
print(classifyNumberLike("abc"))
print(classifyNumberLike(None))
print(classifyNumberLike(None))
